### PR TITLE
Support "when":"true" in 'commandPalette' menus

### PR DESCRIPTION
**Pull request by alvsan09**
Adds a missing check for "true" on contributed "commandPalette" menus

Fixes: #9065

Signed-off-by: Alvaro Sanchez-Leon <alvaro.sanchez-leon@ericsson.com>

#### What it does

Fixes the issue described by #9065, where contributed 'commandPalette' menus including the "when" clause and set to "true"
don't get presented to the user.

#### How to test

In your test extension add the following commands and menus to your package.json

1) Add some commands to a test plugin e.g. 
```
		"commands": [
			{
				"command": "Alvs-extension.dosomething-true",
				"title": "Alvs do something - true",
				"category": "Alvs"
			},
			{
				"command": "Alvs-extension.dosomething-false",
				"title": "Alvs do something - false",
				"category": "Alvs"
			},
			{
				"command": "Alvs-extension.dosomething-undefined",
				"title": "Alvs do something  - undefined",
				"category": "Alvs"
			}
		]
```
2) Add corresponding 'commandPalette' menus to the commands above
e.g.
```
		  "menus": {
			"commandPalette": [ 
			  {
				"command": "Alvs-extension.dosomething-true",
				"when": "true"
			  },
			  {
				"command": "Alvs-extension.dosomething-false",
				"when": "false"
			  },
			  {
				"command": "Alvs-extension.dosomething-undefined"
			  }
			]
		  }
```
3) Display the available commands via 
 >Ctrl-shft-P and filter by 'Alvs'

4) Before this fix you would only see one entry for 'undefined'
    After applying the fix you shall see two entries 'undefined' and 'true'

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by vince-fugnitto — APPROVED** [review]
I confirmed using the following [plugin](https://github.com/vince-fugnitto/vscode-command-palette-when/releases/download/0.0.1/vscode-command-palette-when-0.0.1.vsix) that the commands are properly enabled based on the `when` context:

| Command | When | Result |
|:---|:---|:---|
| Hello World (True) | `'true'` | the command is correctly enabled |
| Hello World (False) | `'false'` | the command is correctly disabled |
| Hello World (Undefined) | `undefined` | the command is correctly enabled |
| Hello World (Dynamic) | `editorHasSelection`| the command is correctly enabled when there is a selection in the editor |

---

In the future I think we can likely re-use the code from [vscode/contextkey.ts](https://github.com/microsoft/vscode/blob/main/src/vs/platform/contextkey/common/contextkey.ts) to handle more advanced use-cases and contexts. If might be low effort to re-use this functionality which is core to the enablement of commands, menus and views.

File: packages/core/src/browser/quick-open/quick-command-service.ts [hunk 1]
```diff
@@ -191,7 +191,10 @@ export class QuickCommandService implements QuickOpenModel, QuickOpenHandler {
         raw.forEach((command: Command) => {
             if (command.label) {
                 const contexts = this.contexts.get(command.id);
-                if (!contexts || contexts.some(when => this.contextKeyService.match(when))) {
+                if (!contexts || contexts.some(when => {
+                    when = when?.trim();
+                    return when === undefined || when === 'true' || this.contextKeyService.match(when);
+                })) {
```
Comment (vince-fugnitto) [inline packages/core/src/browser/quick-open/quick-command-service.ts:197]: `this.contexts` is defined as:

```ts
protected readonly contexts = new Map<string, string[]>();
```

I do not think in this area that when can be `undefined`:

```suggestion
                if (!contexts || contexts.some(when => {
                    when = when.trim();
                    return when === 'true' || this.contextKeyService.match(when);
                })) {
```

Comment (alvsan09) [reply packages/core/src/browser/quick-open/quick-command-service.ts]: Done, Thanks Vince !  A new commit is now available with your proposed change.

**Review by vince-fugnitto — APPROVED** [review]
I confirmed that the updated changes still work correctly with the plugin:
- `true`
- `false`
- `undefined`
- `dynamic` - toggleable through `editorHasSelection`

**Pull request merged**